gdal/frmts/netcdf/netcdflayersg.cpp

```diff
@@ -120,94 +165,137 @@ CPLErr netCDFDataset::LoadSGVarIntoLayer(int ncid, int nc_basevarId)
+
+        this->GrowDim(cdfid, node_coord_dimID,
+            this->GeometryScribe.get_next_write_pos_node_coord() + this->GeometryScribe.getXCBufLength());
+
+
+        if((this->GeometryScribe.getWritableType() == nccfdriver::POLYGON && this->GeometryScribe.getInteriorRingDetected())
+            || this->GeometryScribe.getWritableType() == nccfdriver::MULTILINE || this->GeometryScribe.getWritableType() == nccfdriver::MULTIPOLYGON )
         {
-            poL->FillFeatureFromVar(feat, dimId, featCt);
+            int pnc_dimID = this->GeometryScribe.get_pnc_dimID();
+            this->GrowDim(cdfid, pnc_dimID, this->GeometryScribe.get_next_write_pos_pnc() + this->GeometryScribe.getPNCBufLength());
         }
 
-        feat -> SetFID(featCt);
-        poL->AddSimpleGeometryFeature(feat);
+        this->GeometryScribe.update_ncID(cdfid); // set new CDF ID in case of updates
+        this->GeometryScribe.commit_transaction();
     }
 
-    papoLayers = (netCDFLayer**)CPLRealloc(papoLayers, (nLayers + 1) * sizeof(netCDFLayer *)); 
-    papoLayers[nLayers] = poL;
-    nLayers++;
+    catch(nccfdriver::SG_Exception& sge)
+    {
+        CPLError(CE_Fatal, CPLE_FileIO, "An error occurred while writing the target netCDF File. Translation will be terminated.\n%s", sge.get_err_msg());
+    }
+}
 
-    return CE_None;
+/* Takes an index and using the layer geometry builds the equivalent
+ * OGRFeature.
+ */
+OGRFeature* netCDFLayer::buildSGeometryFeature(size_t featureInd)
+{
+    OGRGeometry * geometry;
+
+    switch(m_simpleGeometry->getGeometryType())
+    {
+        case nccfdriver::POINT:
+            geometry = new OGRPoint;
+            break;
+        case nccfdriver::LINE:
+            geometry = new OGRLineString;
+            break;
+        case nccfdriver::POLYGON:
+            geometry = new OGRPolygon;
+            break;
+        case nccfdriver::MULTIPOINT:
+            geometry = new OGRMultiPoint;
+            break;
+        case nccfdriver::MULTILINE:
+            geometry = new OGRMultiLineString;
+            break;
+        case nccfdriver::MULTIPOLYGON:
+            geometry = new OGRMultiPolygon;
+            break;
+        default:
+            throw nccfdriver::SG_Exception_BadFeature();
+            break;
+    }
+
+    int r_size = 0;
+    std::unique_ptr<unsigned char, std::default_delete<unsigned char[]>> wkb_rep(m_simpleGeometry->serializeToWKB(featureInd, r_size));
+    geometry->importFromWkb(static_cast<const unsigned char*>(wkb_rep.get()), r_size, wkbVariantIso);
+
+    OGRFeatureDefn* defn = this->GetLayerDefn();
+    OGRFeature * feat = new OGRFeature(defn);
+    feat -> SetGeometryDirectly(geometry);
+
+    int dimId = m_simpleGeometry->getInstDim();
+    size_t dim_len = m_simpleGeometry->getInstDimLen();
+    int dim_len_trunc = static_cast<int>(dim_len);
+
+    // Fill fields
+    for(int itr = 0; itr < defn->GetFieldCount() && itr < dim_len_trunc; itr++)
```

Comment:
this loop doesn't make sense to me. FillFeatureFromVar() already takes care about setting all fields of a given feature

Reply:
Oh I didn't know that it already look for all fields; I'll go remove the loop then, thanks :+1: